src/main/java/frc/robot/subsystems/Elevator.java

```diff
@@ -0,0 +1,107 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018-2019 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.subsystems;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.DutyCycleEncoder;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.constants.Constants;
+import frc.robot.constants.ElevatorConstants;
+import frc.robot.controls.Operator;
+import frc.robot.util.MotorFactory;
+
+public class Elevator extends SubsystemBase {
+
+  private final WPI_TalonFX m_elevatorMotor; //do I define these are private final? 
+  private final DigitalInput m_bottomLimitSwitch;
+  private final DigitalInput m_topLimitSwitch; 
+  private final PIDController m_elevatorPID; 
+  private final DutyCycleEncoder m_absoluteSpoolEncoder; 
+
+  public Elevator() {
+    this(
+      MotorFactory.createTalonFX(ElevatorConstants.kElevatorMotor, Constants.kRioCAN),
+      new DigitalInput(-1),
+      new DigitalInput(-1),
+      new DutyCycleEncoder(-1)
+    );
+  }
+
+  public Elevator(WPI_TalonFX motor, DigitalInput bottomSwitch, DigitalInput topSwitch, DutyCycleEncoder absoluteSpoolEncoder) {
+    m_elevatorMotor = motor; 
+    m_bottomLimitSwitch = bottomSwitch; 
+    m_topLimitSwitch = topSwitch;
+    m_absoluteSpoolEncoder = absoluteSpoolEncoder; 
+
+    m_elevatorPID = new PIDController(ElevatorConstants.kElevatorP, ElevatorConstants.kElevatorI, ElevatorConstants.kElevatorD);  
+  }
+
+  public void set(double power){
+    m_elevatorMotor.set(power); 
+  }
+
+  public void stopMotor(){
+    m_elevatorMotor.set(0); 
+  }
+
+  public boolean returnBottomLimSwitchCondition(){
```

Comment:
I would use "get" instead of "return", and I'm not sure "condition" is necessary.

Reply:
same for returnTopLimSwitchCondition()

Reply:
Code change added.